### PR TITLE
Add a config to set TensorRT optimization level

### PR DESCRIPTION
**Pull request by leo0519**
### PR types

New features

### PR changes

APIs

### Description

Since TRT 8.6 EA, the optimization level of building engine can be determined by users with new API [`setBuilderOptimizationLevel`](https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/classnvinfer1_1_1_i_builder_config.html#af51d2c96a3dcc98c345141ff93562e42).

Hence, this PR provides an interface in Paddle Inference for users to set the optimization level.

1. `void SetTensorRtOptimizationLevel(int)` : to set the optimization level
2. `int tensorrt_optimization_level()` : return the optimization level (default is 3 if not set)

This API accepts level in the range [0, 5], and higher optimization level allows the optimizer to spend more time searching for optimization opportunities. The default optimization level is 3.

### CODE REVIEW

File: paddle/fluid/inference/api/analysis_config.cc [hunk 1]
```diff
@@ -855,6 +856,16 @@ void AnalysisConfig::Exp_DisableTensorRtOPs(
 
 void AnalysisConfig::EnableVarseqlen() { trt_use_varseqlen_ = true; }
 
+void AnalysisConfig::SetTRTOptimizationLevel(int level) {
```
Comment (Wangzheee) [inline paddle/fluid/inference/api/analysis_config.cc]: 后续需要加上 api的文档

**Comment by leo0519** [comment]
CI status

- PR-CI-APPROVAL
    ```sh
    2023-09-11 14:47:10 0. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE as EXCLUSIVE, DIST, NIGHTLY, EXCLUSIVE:NIGHTLY or DISTNIGHTLY, or setting parameter SERIAL, or setting TIMEOUT properties.
    2023-09-11 14:47:10 The corresponding lines are as follows:
    2023-09-11 14:47:10 test/ir/inference/CMakeLists.txt
    2023-09-11 14:47:10 +  set_tests_properties(test_trt_optimization_level PROPERTIES TIMEOUT 300)
    2023-09-11 14:47:10 For more information, please refer to:https://github.com/PaddlePaddle/Paddle/wiki/PaddlePaddle-Unit-test-specification
    2023-09-11 14:47:10 There are 1 approved errors.
    ```
- PR-CI-Static-Check
    ```sh
    2023-09-11 17:51:55 Please find RD for approval first, and then find TPM for approval.
    2023-09-11 17:51:55 0. You must have one RD (XiaoguangHu01, jeff41404, lanxianghit or qingqing01) approval for API change.
    2023-09-11 17:51:55  and one TPM approval for API change: 
    2023-09-11 17:51:55  jzhang533/ZhangJun, sunzhongkai588/SunZhongKai, dingjiaweiww/DingJiaWei, Ligoml/LiMengLiu for general APIs.
    2023-09-11 17:51:55  liuTINA0907/LiuShuangQiao for distributed related APIs.
    2023-09-11 17:51:55  leiqing1/LeiQing for inference related APIs.
    2023-09-11 17:51:55 1. You must have one RD (XiaoguangHu01, jeff41404, lanxianghit or qingqing01) approval for API change.
    2023-09-11 17:51:55  and one TPM approval for API change: 
    2023-09-11 17:51:55  jzhang533/ZhangJun, sunzhongkai588/SunZhongKai, dingjiaweiww/DingJiaWei, Ligoml/LiMengLiu for general APIs.
    2023-09-11 17:51:55  liuTINA0907/LiuShuangQiao for distributed related APIs.
    2023-09-11 17:51:55  leiqing1/LeiQing for inference related APIs.
    2023-09-11 17:51:55 2. You must have  one TPM approval for API documents change: 
    2023-09-11 17:51:55  jzhang533/ZhangJun, sunzhongkai588/SunZhongKai, dingjiaweiww/DingJiaWei, Ligoml/LiMengLiu for general API docs.
    2023-09-11 17:51:55  liuTINA0907/LiuShuangQiao for distributed related API docs.
    2023-09-11 17:51:55  leiqing1/LeiQing for inference related API docs.
    2023-09-11 17:51:55 
    2023-09-11 17:51:55 There are 3 approved errors.
    ```

**Review by XieYunshen — DISMISSED** [review]
LGTM for `set_tests_properties(test_trt_optimization_level PROPERTIES TIMEOUT 300)`

**Review by Ligoml — DISMISSED** [review]
后面补一下api文档吧

**Review by Wangzheee — DISMISSED** [review]
LGTM

File: paddle/fluid/pybind/inference_api.cc [hunk 2]
```diff
@@ -927,6 +927,10 @@ void BindAnalysisConfig(py::module *m) {
       .def("tensorrt_explicit_quantization_enabled",
            &AnalysisConfig::tensorrt_explicit_quantization_enabled)
       .def("tensorrt_engine_enabled", &AnalysisConfig::tensorrt_engine_enabled)
+      .def("set_trt_optimization_level",
+           &AnalysisConfig::SetTRTOptimizationLevel)
+      .def("tensorrt_optimization_level",
```
Comment (jzhang533) [inline paddle/fluid/pybind/inference_api.cc:932]: a little bit confused by the names chosen for this setter/getter pair. 
shouldn't it be consistent? i.e.: set_trt_optimization_level/trt_optimization_level, or set_ tensorrt_optimization_level/tensorrt_optimization_level .

Comment (Wangzheee) [reply paddle/fluid/pybind/inference_api.cc:932]: set_trt_optimization_level是指设置optimization的level，trt_optimization_level是指获得当前optimization的level

Comment (leo0519) [reply paddle/fluid/pybind/inference_api.cc:932]: 已將 API 改為 ([commit](https://github.com/PaddlePaddle/Paddle/pull/57031/commits/847b728feaf48a46fac8ea44db4b15f54b8798c7))
- C: `setTensorRtOptimizationLevel` / `tensorrt_optimization_level`
- Python: `set_tensorrt_optimization_level` / `tensorrt_optimization_level`

**Review by jzhang533 — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]

**Review by lanxianghit — APPROVED** [review]
LGTM for internal use only

**Pull request merged**